packages/suite/src/actions/wallet/send/sendFormCardanoActions.ts

```diff
@@ -0,0 +1,218 @@
+import TrezorConnect, { CardanoTxSigningMode } from 'trezor-connect';
+import {
+    getChangeAddressParameters,
+    getTtl,
+    getNetworkId,
+    getProtocolMagic,
+    transformUserOutputs,
+    transformUtxos,
+    formatMaxOutputAmount,
+    loadCardanoLib,
+} from '@wallet-utils/cardanoUtils';
+import * as notificationActions from '@suite-actions/notificationActions';
+import {
+    FormState,
+    UseSendFormState,
+    PrecomposedLevelsCardano,
+    PrecomposedTransactionFinalCardano,
+} from '@wallet-types/sendForm';
+import { Dispatch, GetState } from '@suite-types';
+import { formatNetworkAmount, isTestnet } from '@wallet-utils/accountUtils';
+
+export const composeTransaction =
+    (formValues: FormState, formState: UseSendFormState) =>
+    async (dispatch: Dispatch): Promise<PrecomposedLevelsCardano | undefined> => {
+        const { account, feeInfo } = formState;
+        const changeAddress = getChangeAddressParameters(account);
+        if (!changeAddress || !account.utxo) return;
+
+        const { coinSelection, trezorUtils, CoinSelectionError } = await loadCardanoLib();
+
+        const predefinedLevels = feeInfo.levels.filter(l => l.label !== 'custom');
+        if (formValues.selectedFee === 'custom') {
+            predefinedLevels.push({
+                label: 'custom',
+                feePerUnit: formValues.feePerUnit,
+                blocks: -1,
+            });
+        }
+
+        const utxos = transformUtxos(account.utxo);
+        const outputs = transformUserOutputs(
+            formValues.outputs,
+            account.tokens,
+            formValues.setMaxOutputId,
+        );
+
+        const wrappedResponse: PrecomposedLevelsCardano = {};
+        predefinedLevels.forEach(level => {
+            const options = {
+                ...(level.label === 'custom' ? { feeParams: { a: formValues.feePerUnit } } : {}),
+                debug: true,
```

Comment:
what does debug do? so logging? just asking, I don't really mind as we are now going to merge it hidden behind debug mode.

Reply:
yup, just some debug information so we can troubleshoot why and how the transaction was constructed. We of course don't need it in production I'll remove it.